architecture/index.md

```diff
@@ -1,30 +1,40 @@
 
 
 ### Goals
-- Compose cloud native applications that consist of multiple components and services
-- Provide managed application lifecycle
-- Rapid bootstrapping of applications
-- Fast onboarding of applications to the cloud
-- Supports both existing and new applications
-- Provide APIs to manage your application lifecycle
-- Provide a surface for partners to integrate into the application lifecycle
-
-## Architecture Goals and Constraints
-- Robust delivery automation: Establish continuous delivery practices but also deliver operational tooling.
+- Compose software that consists of multiple components, from multiple repositories.
+- Provide transparency on the software supply chain, including both what makes up the software and how it was built.
+- Provide a way for software teams to release to destinations under the control of their SRE or release engineering team(s).
+- Rapid bootstrapping of new software projects
+- Support both existing and new projects
+- Provide APIs to manage your software lifecycle
+- Provide a surface for partners to integrate, add value
+
+## Architecture Goals
+
 - Just in time scaling: In contrast to “just in case” scaling. The system should be able to scale without capacity reserved ahead of time.
 - Static stability: the overall system continues to work when a dependency is impaired
-- Each subservice can fulfill its primary use cases independently, without relying on  other systems’ availability.
+- Each subservice can fulfill its primary use cases independently, without relying on other systems’ availability.
 - Each sub-service owns its data and logic.
 - Communication among services and participants is always asynchronous.
 - Each sub-service is owned by one team. Ownership does not mean that only one team can change the code, but the owning team has the final decision.
 - Minimize shared infrastructure among sub-services
 - Participants: onboarding new participants, the flexibility to satisfy the technology preferences of a heterogeneous set of participants. Think of this as the ability to easily create an ecosystem and the ability to support that ecosystem’s heterogeneous needs.
 - Security, Privacy, and Governance: Sensitive data is protected by fine-grained access control
 
+## Architecture Constraints
+
+- Our API server is **the kube API server**. Services expose their API as Custom Resource Definitions. This means that requests are necessarily asyncronous. This means that RBAC is implemented the same way across services. In any exceptional case that a service needs to expose its own bespoke HTTP endpoint (like tekton results), use SubjectAccessReviews to ensure RBAC is consistent.
+- **Use tekton** for anything that should be configurable by the user (building, testing, releasing). Expose as much detail via kube resources as possible. Prefer to implement native tasks to perform work on cluster, rather than calling out to third-party services.
```

Comment:
```suggestion
- **Use tekton** for anything that should be extended by the user (building, testing, releasing). Expose as much detail via kube resources as possible. Prefer to implement native tasks to perform work on cluster, rather than calling out to third-party services.
```